### PR TITLE
Implement pureChecks for file handler classes

### LINKED ISSUE #9893: implement pureChecks for File handlers

### PR DESCRIPTION
**Pull request by kimbor**
**Description**:
Implement pureChecks for file handler classes.

**Related issue(s)**:

Fixes #9893

### CODE REVIEW

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/utils/FileServiceUtils.java [hunk 1]
```diff
@@ -63,23 +63,20 @@ public static void validateContent(@NonNull byte[] content, @NonNull FilesConfig
...
-            @Nullable final FileID fileId,
+            @NonNull final FileID fileId,
             @NonNull final ReadableFileStore fileStore,
-            @NonNull final PreHandleContext context,
-            boolean isDelete)
+            @NonNull final PreHandleContext context)
             throws PreCheckException {
         requireNonNull(context);
-
-        if (fileId == null) {
```
Comment (jsync-swirlds) [inline hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/utils/FileServiceUtils.java:80]: I think this specific check was more correct as it was before, a NPE makes dispatch less reliable.

**Review by jsync-swirlds — DISMISSED** [review]
Looks good, with one suggestion.

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/utils/FileServiceUtils.java (hunk 1 above)
Comment (kimbor) [reply hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/utils/FileServiceUtils.java:80]: We are now checking that fileId != null in pureChecks, before we ever get here. It should not ever happen that fileId is null here.

Comment (jsync-swirlds) [reply hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/utils/FileServiceUtils.java:80]: Except that some paths do not call pureChecks (e.g. when a contract or scheduled transaction is gathering keys to be validated).

Comment (jsync-swirlds) [reply hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/utils/FileServiceUtils.java:80]: True, but can we guarantee that will be true indefinitely?  It's not a major concern, I just believe that avoiding NPE, when possible, is preferred.

**Review by povolev15 — DISMISSED** [review]
LGTM, maybe just Joseph suggestion but it is up to you

**Review by iwsimon — APPROVED** [review]
LGTM

**Review by jsync-swirlds — APPROVED** [review]

**Review by iwsimon — APPROVED** [review]

**Pull request merged**